pkg/resource/properties.go

```diff
@@ -257,7 +257,7 @@ func NewPropertyValueRepl(v interface{},
 	switch rk := rv.Type().Kind(); rk {
 	case reflect.Array, reflect.Slice:
 		// If an array or slice, just create an array out of it.
-		var arr []PropertyValue
+		arr := []PropertyValue{}
```

Comment:
please explain this to me so i don't make the same mistake in the future.  i thought these were the same in go :)

Reply:
`var arr []PropertyValue{}` declares a slice of property values. A variable declared using `var` will be zero-initialized (unless you explicitly initialize it). The zero value for a slice is `nil`.

`arr := []PropertyValue{}` declares and initializes a variable `arr` with an empty but non-`nil` slice.